keps/sig-instrumentation/1748-pod-resource-metrics/README.md

```diff
@@ -409,11 +416,198 @@ enhancement:
+  - [ ] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name:
+    - Components depending on the feature gate:
+  - [x] Other
+    - Describe the mechanism: A metrics collector may scrape the `/metrics/resources` endpoint of all schedulers, as long as the scheduler exposes metrics of the required stability level.
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+* **Does enabling the feature change any default behavior?**
+
+Scraping these metrics does not change behavior of the system.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+
+Administrators may stop scraping the endpoint, which will mean the metrics are not available.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+
+Metrics will start getting collected.
+
+* **Are there any tests for feature enablement/disablement?**
+
+As an opt-in metrics endpoint enablement is tested from our integration tests.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+
+This cannot impact running workloads unless an unlikely performance issue is triggered due to
+excessive scraping of the scheduler metrics endpoints (which is already possible today).
+
+Since the new metrics are proportionally less than the metrics an apiserver or node exposes,
+it is unlikely that scraping this endpoint would break a metrics collector.
+
+* **What specific metrics should inform a rollback?**
+
+Excessive CPU use from the Kube scheduler when metrics are scraped at a reasonable rate,
+although simply disabling optional scraping while waiting for the bug to be fixed would be
+a more reasonable path.
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+
+Does not apply.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs,
+fields of API types, flags, etc.?**
+
+No.
+
+### Monitoring Requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+
+This would be up to the metrics collector component whose API is not under the
+scope of the Kubernetes project. Some third party software may use these metrics
+as part of a control loop or visualization, but that is entirely up to the metrics
+collector.
+
+Administrators and visualization tools are the primary target of these metrics and
+so polling and canvassing of Kube distributions is one source of feedback.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine
+the health of the service?**
+  - [ ] Metrics
+    - Metric name:
+    - [Optional] Aggregation method:
+    - Components exposing the metric:
+  - [x] Other (treat as last resort)
```

Comment:
Can you list any specific ones?

Reply:
In this case I would probably look at scheduler up / down from the scraping endpoint, then scheduler cpu.  Is that what you're looking for?

Reply:
Sure, or you can just say: Existing scheduler SLIs.

Reply:
nit: you could just remove the template text and put the answer